packages/kbn-shared-ux-services/src/context.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { FC, createContext, useContext } from 'react';
+
+import type { SharedUxServices } from './types';
+import { stubServicesFactory } from './services';
+
+// The React Context used to provide the services to the SharedUX components.
+const SharedUxServicesContext = createContext<SharedUxServices>(stubServicesFactory());
```

Comment:
Why is this using `stubServiceFactory`? I know I had asked this question before, but I'm still failing to understand how is the services package wired with actual services from the `shared_ux` plugin?

Reply:
It's using `stubServicesFactory` to populate a default set of services on this instance of the React Context.  Passing nothing here would mean we'd constantly have to check for undefined.

The hooks in this file refer to this Context API.  When a consumer of our components/services wants to use them, they have two choices:

1. use the pre-wired Context, or
2. wire it up themselves with their own `services` value.

## Pre-wired Context

The Shared UX plugin [imports](https://github.com/elastic/kibana/pull/127546/files#diff-13f98fdc7f2db7c405dcdd95036867aefae3fbe97eed26a5ca20faa2f395cdb3R11) the `SharedUxServicesProvider`  and populates the `services` using a [`servicesFactory`](https://github.com/elastic/kibana/pull/127546/files#diff-c620d5e808d0eb490db0b5d06c38f5bf3b572152e850d7739b11e12f3ca4f58bR22) that uses Kibana dependencies to fulfill the contract.  _That's_ how consumers of the hooks get to the Kibana version of these services: the `SharedUxServicesProvider` _provides_ the API to those hooks, (and thus the components).

## Wiring Manually

You can consider the code in the Shared UX plugin to be a _manual_ wiring of the Provider for components, but Storybook and Jest are also examples:

- Storybook: https://github.com/elastic/kibana/pull/127546/files#diff-900a36a9902b8f2a2f3d20a02b6ad9ed81bf4a5cdb30aa5378f01c593c038d70R20
- Jest: https://github.com/elastic/kibana/pull/127546/files#diff-1e40ad313fcab37f0e2c67b67f113c196e2182bcd5aa4278e1f8f1a4f876e3b7R29

Does this help?

Reply:
I can make that adjustment, sure.

Reply:
Thanks for the detailed explanation, much clearer now. Also agree with Spencer, wiring it up with stubbed version seems a bit error-prone.